src/Symfony/Component/HttpKernel/HttpCache/Store.php

```diff
@@ -350,13 +350,13 @@ private function doPurge($url)
      *
      * @param string $key The store key
      *
-     * @return string The data associated with the key
+     * @return string|null The data associated with the key
      */
     private function load($key)
     {
         $path = $this->getPath($key);
 
-        return file_exists($path) ? file_get_contents($path) : false;
+        return file_exists($path) ? file_get_contents($path) : null;
```

Comment:
As suggested by @Tobion in https://github.com/symfony/symfony/pull/32993#discussion_r311219589

Reply:
file_get_contents can also return false that we should cast to null. this will otherwise break the type hints later

Reply:
Are we talking about the file behind `$path` being deleted between the `file_exists()` and the `file_get_contents()` call or are there other reasons why `file_get_contents` could return `false` in this case?

Reply:
permissions, race conditions and more probably

Reply:
So, something like that?

```php
return file_exists($path) && false !== $contents = file_get_contents($path) ? $contents : null;
```

Reply:
lgtm

Reply:
#33016